### PR TITLE
types: fix cache types resolving to `never`

### LINKED ISSUE #7160: interaction type narrowing left with never type
### Issue description

When handling an interaction, if I narrow the type using `Interaction.inGuild()`, the type is narrowed to one such that further calls to `Interaction.inRawGuild()` or `Interaction.inCachedGuild()` will end up with the type `never`. 
My naive fix suggestion is to simply change the return type of `Interaction.inGuild()` from `this is Interaction<'present'>` to `this is Interaction<'cached' | 'raw'>`. This would fix my issue, but I don't know what the implications of this change are.
Hopefully I haven't misunderstood that `inGuild()` and `inCachedGuild()` are not mutually exclusive.

As a side note: 'present' is a bit weird of a name, I'm guessing that it means that it exists?

### Code sample

```typescript
import { CommandInteraction } from "discord.js";

const handleAsync = async (i: CommandInteraction) => {
    if (!i.inGuild()) {
        return;
    }
    // i is now CommandInteraction<"present"> & CommandInteraction<CacheType>

    if (i.inRawGuild()) {
        // i is never
        return;
    }

    if (i.inCachedGuild()) {
        // i is never
        return;
    }
}
```

### discord.js version

13.4.0

### Node.js version

16.13.1

### Operating system

WSL 2.0 on Windows 11

### Priority this issue should have

Low (slightly annoying)

### Which partials do you have configured?

No Partials

### Which gateway intents are you subscribing to?

GUILDS

### I have tested this issue on a development release

_No response_

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

Closes #7160

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

**Comment by suneettipirneni** [comment]
@yi-fan-song this seems to resolve the use-case given in your issue ticket, and it passes all tests. However can you try this branch out just to see if it broke anything on your end?

File: typings/index.d.ts [hunk 1]
```diff
@@ -722,7 +722,7 @@ export interface ApplicationCommandInteractionOptionResolver<Cached extends Cach
 
 export class CommandInteraction<Cached extends CacheType = CacheType> extends BaseCommandInteraction<Cached> {
   public options: Omit<CommandInteractionOptionResolver<Cached>, 'getMessage' | 'getFocused'>;
-  public inGuild(): this is CommandInteraction<'present'>;
+  public inGuild(): this is CommandInteraction<'cached' | 'raw'>;
```
Comment (imranbarbhuiya) [inline typings/index.d.ts:725]: In the above 2 line you used `'raw' | 'cached'` so ig you should use the same here too.

File: typings/index.d.ts [hunk 2]
```diff
@@ -735,7 +735,7 @@ export class AutocompleteInteraction<Cached extends CacheType = CacheType> exten
   public commandName: string;
   public responded: boolean;
   public options: Omit<CommandInteractionOptionResolver<Cached>, 'getMessage'>;
-  public inGuild(): this is AutocompleteInteraction<'present'>;
+  public inGuild(): this is AutocompleteInteraction<'cached' | 'raw'>;
```
Comment (imranbarbhuiya) [inline typings/index.d.ts:738]: Here too

**Review by imranbarbhuiya — CHANGES_REQUESTED** [review]

**Review by iCrawl — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Comment by yi-fan-song** [comment]
> @yi-fan-song this seems to resolve the use-case given in your issue ticket, and it passes all tests. However can you try this branch out just to see if it broke anything on your end?

got the time to try it, it's perfect on my end, thanks

**Pull request merged**